lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -160,7 +160,9 @@ def repo_checkout_bash_command(repo, url)
         git fetch origin
         git fetch --prune origin +refs/tags/*:refs/tags/*
         git reset --merge
-        git checkout #{branch}
+        if ! git checkout #{branch}; then
+          git checkout -b #{branch} origin/#{branch}
```

Comment:
I'm also still not clear in what scenario `git checkout XX` is not equivalent to `git checkout -b XX origin/XX` given that `origin/XX` is a valid ref.